### PR TITLE
fix: Create DraftOrder flow

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Fix nested modal error with products page that would cause claim to fail
- Support for changing regions in draft orders

**How**
- Move price calculations for new variants to the draft order flow rather than the sub-modal products screen
- Update field array when updating quantities
- Update prices when changing regions in the new-order form

Fixes CORE-1082

### CODE REVIEW

**Review by olivermrbl — CHANGES_REQUESTED** [review]
I can't seem to update the already added items. Do you know what might cause this? Adding a screen recording below.

https://user-images.githubusercontent.com/59018053/217010046-eab6a17f-2f4f-48f3-aadd-1d2a9bdf5de7.mov

File: src/domain/orders/details/rma-sub-modals/products.tsx [hunk 1]
```diff
@@ -11,7 +11,6 @@ import { LayeredModalContext } from "../../../../components/molecules/modal/laye
 import Table from "../../../../components/molecules/table"
```
Comment (olivermrbl) [inline src/domain/orders/details/rma-sub-modals/products.tsx:11]: comment: Also fixed in #881 @kasperkristensen

**Comment by olivermrbl** [comment]
@kasperkristensen Would you be able to have a look at this? 🙏

File: src/domain/orders/new/components/items.tsx [hunk 2]
```diff
@@ -32,19 +32,27 @@ const Items = () => {
...
 
-  const addItem = (variants: PricedVariant[]) => {
+  const addItem = async (variants: ProductVariant[]) => {
     const ids = fields.map((field) => field.variant_id)
+
     const itemsToAdd = variants.filter((v) => !ids.includes(v.id))
 
+    const variantIds = itemsToAdd.map((v) => v.id)
+
+    const { data } = (await Medusa.variants.list({
```
Comment (kasperkristensen) [inline src/domain/orders/new/components/items.tsx:49]: nit (non-blocking): I think we should look to phase out the old API service, so I am not too keen on adding the use of it to a new file. Could we use medusajs instead?

**Review by kasperkristensen — APPROVED** [review]
LGTM!

File: src/domain/orders/new/components/items.tsx (hunk 2 above)
Comment (olivermrbl) [reply src/domain/orders/new/components/items.tsx:49]: @pKorsholm Do you have any comments on this? :)

Comment (pKorsholm) [reply src/domain/orders/new/components/items.tsx]: changed it 😄

**Review by olivermrbl — APPROVED** [review]
LGTM! 😍

**Pull request merged**